ext/intl/formatter/formatter_format.c

```diff
@@ -53,23 +53,23 @@ PHP_FUNCTION( numfmt_format )
 	/* Fetch the object. */
 	FORMATTER_METHOD_FETCH_OBJECT;
 
-	if(type == FORMAT_TYPE_DEFAULT) {
-		if(Z_TYPE_P(number) == IS_STRING) {
-			convert_scalar_to_number_ex(number);
-		}
-
-		if(Z_TYPE_P(number) == IS_LONG) {
-			/* take INT32 on 32-bit, int64 on 64-bit */
-			type = (sizeof(zend_long) == 8)?FORMAT_TYPE_INT64:FORMAT_TYPE_INT32;
-		} else if(Z_TYPE_P(number) == IS_DOUBLE) {
-			type = FORMAT_TYPE_DOUBLE;
-		} else {
-			type = FORMAT_TYPE_INT32;
-		}
+	if(Z_TYPE_P(number) != IS_ARRAY) {
+		convert_scalar_to_number_ex(number);
+	} else {
+		convert_to_long(number);
```

Comment:
I'd suggest treating this as an error condition instead.

Reply:
I don't think we can break BC here for PHP-7.3 and 7.4 (passing an array is currently supported, and the docs even implicitly support that: "Can be integer or float, other values will be converted to a numeric value."). I think changing this for master, would be prudent, though.